### PR TITLE
Check for uncommitted output

### PR DESCRIPTION
**Pull request by pimterry**
Connects to #573

With this change, if you commit any source files without also committing the build output, Travis will build, spot it, and fail.

This also conveniently means we're now actually running the build itself in CI, so Travis catch at least obvious compile-breaking errors in future.
---- Autogenerated Waffleboard Connection: Connects to #573

### CODE REVIEW

**Comment by pimterry** [comment]
Ha, the build then fails, because we've generated a package-lock.json in travis that we haven't committed. Travis using npm@5 seems risky to me, given that we're not happy its stable enough to use locally, so I've bumped it down to 4.

File: automation/catch-uncommitted-output [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#!/bin/sh
```
Comment (lekkas) [inline automation/catch-uncommitted-output]: Can we rename the file to `catch-uncommitted-output.sh` to give a hint on its type?

File: automation/catch-uncommitted-output [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+set -e
+
+npm run build
+
+if ! git diff HEAD --exit-code ; then
+	echo '\n** Uncommitted changes found after build - FAIL **'
+	exit 1
+elif test -n "$(git ls-files --exclude-standard --others)" ; then
```
Comment (lekkas) [inline automation/catch-uncommitted-output]: It'd be nice if we could avoid duplication of `git ls-files --exclude-standard --others` here

File: automation/catch-uncommitted-output (hunk 1 above)
Comment (pimterry) [inline automation/catch-uncommitted-output]: Good idea, done

File: automation/catch-uncommitted-output (hunk 2 above)
Comment (pimterry) [inline automation/catch-uncommitted-output]: Agree, done

**Pull request merged**